include/net/net_ip.h

```diff
@@ -179,7 +179,11 @@ struct net_addr {
 extern const struct in6_addr in6addr_any;
 extern const struct in6_addr in6addr_loopback;
 
+/* Defined by POSIX. INET6_ADDRSTRLEN accounts for mapped IPv4 addresses. */
+#define INET_ADDRSTRLEN 16
 #define INET6_ADDRSTRLEN 46
+
+/* These are for internal usage of the stack */
 #define NET_IPV6_ADDR_LEN sizeof("xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx")
```

Comment:
And here I would just go (but also don't insist, decide yourelf) with:
```
#define NET_IPV4_ADDR_LEN INET_ADDRSTRLEN
#define NET_IPV6_ADDR_LEN INET6_ADDRSTRLEN
```

Reply:
I would like to keep the original sizeof() style define. As for IPv6 we definitely want to use the shorter sizeof() version.

Reply:
> And here I would just go (but also don't insist, decide yourelf)

That would be next possible step, which requires discussion (as started in https://github.com/zephyrproject-rtos/zephyr/pull/9818).

We can't do it right away, because NET_IPV6_ADDR_LEN != INET6_ADDRSTRLEN.

Reply:
In case of all-POSIX approach would it also make sense to flip wrappers in include/net/socket.h, i.e. that the primary implementation is in inet_ntop/pton()?

Reply:
> We can't do it right away, because NET_IPV6_ADDR_LEN != INET6_ADDRSTRLEN

Then it makes sense to reuse the original sizeof() style defines for INET_ADDRSTRLENs to keep the semantics intact (as @jukkar suggests) and defer INET6_ADDRSTRLEN=46 until we are fully compliant with IPv6 mapped IPv4 addresses (in case there's such a need).

Reply:
> defer INET6_ADDRSTRLEN=46

I don't do anything about INET6_ADDRSTRLEN, it's already there, and as 46. And 46 comes from http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/netinet_in.h.html , so defining it to something else is petty, but a violation, we need to think about it more. (#9818 is for that thinking/discussion, and there's no hurry with it I guess.)